external_api/lib/dictionary_utils.py

```diff
@@ -25,45 +47,62 @@ def dictionary_update_languages(dictionary_name, new_language_list):
         raise APIHandlerException('No new languages provided. Nothing changed.')
 
     # fetch all words
-    word_list = get_dictionary_unique_words(dictionary_name)
-    if not word_list:
-        raise APIHandlerException('This dictionary does not have any records. Please verify the dictionary name')
+    values = get_entity_unique_values(entity_name=entity_name)
+    if not values:
+        raise APIHandlerException('This entity does not have any records. Please verify the entity name')
 
     records_to_create = []
     for language_script in languages_added:
         # create records for all words
-        for word in word_list:
+        for value in values:
             records_to_create.append({
-                'word': word,
+                'word': value,
```

Comment:
The left-hand side can be 'value' as it is passed to Datastore API. It will avoid creating a new term for Datastore